### PR TITLE
Update `new Error` calls in Container class to use fluid error classes

### PR DESCRIPTION
**Pull request by markfields**
One caveat - I'm adding more usages of `UsageError` here, which may be updated/renamed in accordance with pending discussions about error shapes. But it's already used in the file so I figure no real harm in it, and these new cases match the semantics elsewhere in the file (if I'm interpreting these codepaths properly).

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ export enum ConnectionState {
 export async function waitContainerToCatchUp(container: Container) {
     // Make sure we stop waiting if container is closed.
     if (container.closed) {
-        throw new Error("Container is closed");
+        throw new UsageError("refusingToWaitForClosedContainer");
```
Comment (pleath) [inline packages/loader/container-loader/src/container.ts:175]: Is the use of strings without spacing required by the new API?

Comment (pleath) [reply packages/loader/container-loader/src/container.ts:175]: Or is it desirable for some other reason?

Comment (vladsud) [reply packages/loader/container-loader/src/container.ts:175]: I think UsageError and usage of it here is wrong. fluidErrorCode != message to developer, who's locally debugging the issue. fluidErrorCode should be renamed to be callSite, and I think in future may be transitioned to use tags (similar to asserts, to save bundle size).
Message should be pretty clear (to English-speaking developer) on why he/she is getting this error. It has to be in plain English :)

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -623,7 +617,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     this.logger.sendErrorEvent({
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:617]: (nit) I'd remove this event. IMHO it does not give anything - errors should be logged by outer scenario, and error raised should have enough info to point to this code (origin of error)

File: packages/loader/container-loader/src/container.ts [hunk 3]
```diff
@@ -1827,7 +1821,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     ) {
         const codeDetails = this.getCodeDetailsFromQuorum();
         if (codeDetails === undefined) {
-            throw new Error("pkg should be provided in create flow!!");
+            // pkg should be provided in create flow!!
+            throw new GenericError("instantiateContextDetachedMissingCodeDetails");
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:1825]: Is this usage error?

**Comment by markfields** [comment]
In light of #8929, closing this.  We still need to update these errors, but will just start from scratch.
